src/services/dapp.service.ts

```diff
@@ -0,0 +1,18 @@
+import { DappId } from '../model/general.types'
+import { SwarmExtension } from '../swarm-api/swarm-extension'
+import { dappUrlToId } from './fdp-storage/fdp-storage.utils'
+import { Storage } from './storage/storage.service'
+
+export class DappService {
+  private storage = new Storage()
+
+  public async getDappId(dappUrl: string): Promise<DappId> {
+    const { extensionId } = await this.storage.getSwarm()
+
+    const swarmExtension = new SwarmExtension(extensionId)
+
+    const { beeApiUrl } = await swarmExtension.beeAddress()
+
+    return dappUrlToId(dappUrl, beeApiUrl)
```

Comment:
if the beeApiUrl is localhost, the dappId can be in the subdomain as well in a CID format already.